githubapp/client_creator.go

```diff
@@ -132,6 +134,13 @@ func WithClientUserAgent(agent string) ClientOption {
 	}
 }
 
+// WithClientCache sets the http cache for all created clients
+func WithClientCache(cache httpcache.Cache) ClientOption {
```

Comment:
After thinking about this more, I believe the user actually needs to provide a `func() httpcache.Cache` instead of a single instance. Cache keys appear to use only the URL and method, meaning they don't account for authentication. If we share a single cache between all clients, I think we could easily expose private repositories or other information to clients that don't have permissions to see them.